controller/eks-cluster-config-handler.go

```diff
@@ -141,6 +148,52 @@ func (h *Handler) recordError(onChange func(key string, config *eksv1.EKSCluster
 	}
 }
 
+func removeErrorMetadata(message string) (string, error) {
+
+	// failure message
+	type RespMetadata struct {
+		StatusCode int    `json:"statusCode"`
+		RequestID  string `json:"requestID"`
+	}
+
+	type Message struct {
+		RespMetadata  RespMetadata `json:"respMetadata"`
+		ClusterName   string       `json:"clusterName"`
+		Message_      string       `json:"message_"`
+		NodegroupName string       `json:"nodegroupName"`
+	}
+
+	// failure message with no meta
+	type FailureMessage struct {
+		ClusterName   string `json:"clusterName"`
+		Message_      string `json:"message_"`
+		NodegroupName string `json:"nodegroupName"`
+	}
+
+	// Remove the first line of the message because it usually contains the name of an Amazon EKS error type that
+	// implements Seralizable (ex: ResourceInUseException). That name is unpredictable depending on the error. We
+	// only need cluster name, message, and node group.
+	index := strings.Index(message, "{")
+	message = message[index:]
```

Comment:
`Strings.Index` will return `-1` if the substring isn't found, in which case `message[index:]` would panic. Will all error messages pass through this function?

Reply:
Should check for -1 and return error if true.